plugin/trino-kafka-event-listener/src/main/java/io/trino/plugin/eventlistener/kafka/KafkaEventListener.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.eventlistener.kafka;
+
+import com.google.inject.Inject;
+import io.airlift.log.Logger;
+import io.trino.plugin.eventlistener.kafka.metrics.KafkaEventListenerJmxStats;
+import io.trino.plugin.eventlistener.kafka.producer.KafkaProducerFactory;
+import io.trino.spi.eventlistener.EventListener;
+import io.trino.spi.eventlistener.QueryCompletedEvent;
+import io.trino.spi.eventlistener.QueryCreatedEvent;
+import io.trino.spi.eventlistener.SplitCompletedEvent;
+import org.weakref.jmx.Flatten;
+import org.weakref.jmx.Managed;
+
+public class KafkaEventListener
+        implements EventListener
+{
+    private static final Logger LOG = Logger.get(KafkaEventListener.class);
+
+    private final KafkaEventListenerJmxStats stats = new KafkaEventListenerJmxStats();
+    private final boolean publishCreatedEvent;
+    private final boolean publishCompletedEvent;
+    private final boolean isAnonymizationEnabled;
+    private KafkaEventPublisher kafkaPublisher;
+
+    @Inject
+    public KafkaEventListener(KafkaEventListenerConfig config, KafkaProducerFactory producerFactory)
+            throws Exception
+    {
+        publishCreatedEvent = config.getPublishCreatedEvent();
+        publishCompletedEvent = config.getPublishCompletedEvent();
+        isAnonymizationEnabled = config.isAnonymizationEnabled();
+
+        try {
+            kafkaPublisher = new KafkaEventPublisher(config, producerFactory, stats);
+        }
+        catch (Exception e) {
+            if (!config.getTolerateStartupFailure()) {
```

Comment:
nit: maybe `config.isTerminateOnInitializationFailureEnabled()` with corresponding `kafka-event-listener.terminate-on-initialization-failure.enabled`

Reply:
Renamed the config to `kafka-event-listener.terminate-on-initialization-failure`